File: gymapp/gymapp/page/gym_member_profile/gym_member_profile.js
frappe.pages['gym-member-profile'].on_page_load = function (wrapper) {
    let page = frappe.ui.make_app_page({
        parent: wrapper,
        title: 'Member Profile',
        single_column: true
    });

    frappe.call({
        method: "gymapp.gymapp.page.gym_member_profile.gym_member_profile.get_member_data",
        args: {
            member: frappe.session.user
        },
        callback: function (r) {
            if (r.message) {
                let member_data = r.message;

                $(wrapper).html(`
                    <div class="profile-container">
                        <div class="plan">
                            <img src="${member_data.img}" alt="Profile Image" class="profile-image">
                            <h2>Welcome, ${member_data.member_name}</h2>
                    
                            <p class = "heading">Here are your details and plan:</p>
                            <p><b>Active Plan:</b> ${member_data.active_plan}</p>
                            <p><b>Gym Join Date:</b> ${member_data.start_date}</p>
                            <p><b>Remaining Days:</b> ${member_data.remaining_days}</p>
                            <p><b>Gym Amount:</b> ${member_data.gym_amount}</p>
                        </div>

                        <div class ="sub">
                            <p class="heading">Here are your subscription details:</p>
                            <p><b>Trainer Name:</b> ${member_data.trainer_name}</p>
                            <p><b>Sub Start:</b> ${member_data.sub_start}</p>
                            <p><b>Remaining Day:</b> ${member_data.sub_remaining_day}</p>
                            <p><b>Total Hours:</b> ${member_data.total_hours}</p>
                            <p><b>Total Rate:</b> ${member_data.total_rate}</p>
                            

                        </div>

                        <div class="button-container">
                            <button id="btn-book-locker" class="btn-bookLocker">Book Locker</button>
                            <button id="btn-trainer-subscription" class="btn-trainerSubscription">Trainer Subscription</button>
                            <button id="btn-book-class" class="btn-bookClass">Book Class</button>
                        </div>
                    </div>
                `);

                $('#btn-book-locker').click(function () {
                    window.location.href = 'http://192.168.167.121:8003/locker-booking/new';
                });

                $('#btn-trainer-subscription').click(function () {
                    window.location.href = 'http://192.168.167.121:8003/trainer-subscription-form/new';
                });

                $('#btn-book-class').click(function () {
                    window.location.href = 'http://192.168.167.121:8003/book-class/new';
                });
            }
        }
    });
};

File: gymapp/gymapp/page/gym_member_profile/gym_member_profile.py
import frappe
from frappe.utils import today, date_diff

@frappe.whitelist()
def get_member_data(member):
    member_doc = frappe.get_doc("Gym Member", {"email": member})

    # default value ko Initialize kr rha hai aur ye zaruri hai data show krne k liye.
    plan = None 
    tra = None

    if member_doc.active_plan:
        plan = frappe.get_doc("Gym Membership", member_doc.active_plan)
        remaining_days = date_diff(plan.end_date, today())

    if member_doc.assign_trainer:
        tra = frappe.get_doc("Trainer Subscription", member_doc.assign_trainer)
        sub_remaining_day = date_diff(tra.end_date, today())

    return {
        "member_name": member_doc.full_name,
        "img": member_doc.profile_image,

        "start_date": plan.start_date,
        "remaining_days": remaining_days or 0,
        "active_plan": plan.membership_plan if plan else "None",
        "gym_amount" : plan.amount,

        "trainer_name": tra.trainer or "None",
        "sub_start" : tra.start_date,
        # "sub_end": tra.end_date,
        # "total_days": tra.total_days,
        "sub_remaining_day": sub_remaining_day,
        "total_hours": tra.total_hours,
        "total_rate": tra.total_rate,
        # "sub_amount" : trainer_amount_paid,
        # "trainer_contact": member_doc.allocated_trainer_contact or "None"
    }